test/de/willuhn/jameica/hbci/server/TestVerwendungszweckUtil.java

```diff
@@ -2,315 +2,679 @@
-      "123456789012345678901234567890 ",
-      null,
-      "d",
-      " ",
-      "123456789012345678901234567"
-    };
-    
-    String[] result = VerwendungszweckUtil.rewrap(27,test);
-    
-    Assert.assertTrue(Arrays.equals(new String[]{"123456789012345678901234567",
-                                                 "890123456789012345678901234",
-                                                 "567890d12345678901234567890",
-                                                 "1234567"},
-                                    result));
-  }
-  
-  /**
-   * Testet das Parsen der Tags.
-   * @throws Exception
    */
-  @Test
-  public void testParse001() throws Exception
+  @Nested
+  class Rewrap
   {
-    String [] test =
-    {
-        "SVWZ+Ein komischer.Verwendungszweck ",
-        "auf mehreren Zeilen ",
-        "Hier kommt nochwas",
-        "ABWA+Das ist ein..Test//Text ",
-        "mit Zeilen-Umbruch"
-    };
-    
-    Map<Tag,String> map = VerwendungszweckUtil.parse(test);
-    Assert.assertEquals("SVWZ falsch","Ein komischer.Verwendungszweck auf mehreren Zeilen Hier kommt nochwas",map.get(Tag.SVWZ));
-    Assert.assertEquals("ABWA falsch","Das ist ein..Test//Text mit Zeilen-Umbruch",map.get(Tag.ABWA));
-  }
+    @Test
+    void istNullmitParameterNull() // TODO besser IllegalArgumentException?
+    {
+      assertNull(VerwendungszweckUtil.rewrap(5, (String[]) null));
+    }
 
-  /**
-   * Testet das Parsen der Tags.
-   * @throws Exception
-   */
-  @Test
-  public void testParse002() throws Exception
-  {
-    String [] test =
-    {
-        "SVWZ+Das folgende Tag gibts nicht",
-        "Fooo",
-        "ABCD+Gehoert zum Verwendungszweck"
-    };
-    
-    Map<Tag,String> map = VerwendungszweckUtil.parse(test);
-    Assert.assertEquals("SVWZ falsch","Das folgende Tag gibts nichtFoooABCD+Gehoert zum Verwendungszweck",map.get(Tag.SVWZ));
-  }
+    @Test
+    void leeresArray()
+    {
+      String[] test = {};
+      String[] result = VerwendungszweckUtil.rewrap(3, test);
+      assertArrayEquals(new String[]{}, result);
+    }
 
-  /**
-   * Testet das Parsen der Tags.
-   * @throws Exception
-   */
-  @Test
-  public void testParse003() throws Exception
-  {
-    String [] test =
-    {
-        "SVWZ+Das folgende Tag gibts nicht",
-        "Fooo ",
-        "ABCD+Gehoert zum Verwendungszweck",
-        "EREF+Aber hier kommt noch was"
-    };
-    
-    Map<Tag,String> map = VerwendungszweckUtil.parse(test);
-    Assert.assertEquals("SVWZ falsch","Das folgende Tag gibts nichtFooo ABCD+Gehoert zum Verwendungszweck",map.get(Tag.SVWZ));
-    Assert.assertEquals("EREF falsch","Aber hier kommt noch was",map.get(Tag.EREF));
-  }
+    @Test
+    void nurNullImArray() // TODO besser {} als Ergebnis?
+    {
+      String[] test = {
+              null,
+              null,
+              null,
+              null
+      };
+      String[] result = VerwendungszweckUtil.rewrap(3, test);
+      assertArrayEquals(test, result);
```

Comment:
Das hier ist abweichendes Verhalten: Der Input ist invalide, aber wird 1:1 so durchgereicht.
Aus meiner Sicht sind zwei mögliche Alternativen denkbar:

1. identisches Verhalten wie bei `istNullmitParameterNull()` oben -- egal, was das Ergebnis der Diskussion ist, so auch hier damit umgehen
2. wie in `nullImArrayIgnorieren()` überprüft, die `null`-Werte herausfiltern -- somit wäre der Rest hier ein leeres Array